centml/compile.py

```diff
@@ -0,0 +1,53 @@
+import builtins
+from typing import Callable, Dict, Optional, Union
+
+import torch
+
+from centml.compiler.backend import centml_dynamo_backend
+from centml.compiler.config import OperationMode, settings
+from centml.compiler.prediction.backend import centml_prediction_backend, gauge
+
+
+def compile(
+    model: Optional[Callable] = None,
+    *,
+    fullgraph: builtins.bool = False,
+    dynamic: Optional[builtins.bool] = None,
+    mode: Union[str, None] = None,
+    options: Optional[Dict[str, Union[str, builtins.int, builtins.bool]]] = None,
+    disable: builtins.bool = False,
+) -> Callable:
+
+    if settings.MODE == OperationMode.REMOTE_COMPILATION:
+        # Return the remote-compiled model
+        compiled_model = torch.compile(
+            model,
+            backend=centml_dynamo_backend,  # Compilation backend
+            fullgraph=fullgraph,
+            dynamic=dynamic,
+            mode=mode,
+            options=options,
+            disable=disable,
+        )
+        return compiled_model
+    else:
```

Comment:
lets use else if and check if the MODE is PREDICTION.

else case should throw an except saying the MODE is not supported.